modin/backends/pandas/query_compiler.py

```diff
@@ -877,12 +908,21 @@ def reduce_func(df, **kwargs):
 
     quantile_for_single_value = ReductionFunction.register(pandas.DataFrame.quantile)
     mad = ReductionFunction.register(pandas.DataFrame.mad)
-    to_datetime = ReductionFunction.register(
-        lambda df, *args, **kwargs: pandas.to_datetime(
-            df.squeeze(axis=1), *args, **kwargs
-        ),
-        axis=1,
-    )
+
+    def to_datetime(self, *args, **kwargs):
+        if len(self.columns) == 1:
+            return MapFunction.register(
+                # to_datetime has inplace side effects, see GH#3063
+                handle_readonly_buffer(
```

Comment:
We should probably pass `display_name="to_datetime" in `handle_readonly_buffer` to correctly display the name of being executed method with progress base?

Reply:
Yes, good catch!